ui/src/app/base/components/NodeName/NodeName.tsx

```diff
@@ -10,15 +10,17 @@ import NodeNameFields from "./NodeNameFields";
 import FormikForm from "app/base/components/FormikForm";
 import { useCanEdit } from "app/base/hooks";
 import { hostnameValidation } from "app/base/validation";
+import type { Controller } from "app/store/controller/types";
 import type { Device } from "app/store/device/types";
 import type { Domain, DomainMeta } from "app/store/domain/types";
 import type { Machine } from "app/store/machine/types";
 import type { SimpleNode } from "app/store/types/node";
+import { nodeIsController } from "app/store/utils";
 
 export type Props = {
   editingName: boolean;
   // Machines and devices can edit their name, but no controllers.
-  node: Machine | Device | null;
+  node: Machine | Device | Controller | null;
```

Comment:
We also have a `Node` type that covers these three:

https://github.com/canonical-web-and-design/maas-ui/blob/main/ui/src/app/store/types/node.ts#L209

Reply:
Ah yeah good point 😄